src/app/frontend/deploy/deploy_controller.js

```diff
@@ -80,44 +57,22 @@ export default class DeployController {
     /** @private {!md.$dialog} */
     this.mdDialog_ = $mdDialog;
 
-    /** @private {boolean} */
-    this.isDeployInProgress_ = false;
+    /** @export {boolean}*/
+    this.isDeployInProgress = false;
+
+    /** @export {!Object}*/
+    this.detail = {};
   }
 
   /**
-   * Deploys the application based on the sate of the controller.
+   * Notifies the child scopes to call their deploy methods.
    *
    * @export
    */
-  deploy() {
-    // TODO(bryk): Validate input data before sending to the server.
-
-    /** @type {!backendApi.AppDeploymentSpec} */
-    let appDeploymentSpec = {
-      containerImage: this.containerImage,
-      isExternal: this.isExternal,
-      name: this.name,
-      description: this.description,
-      portMappings: this.portMappings.filter(this.isPortMappingEmpty_),
-      replicas: this.replicas,
-      namespace: this.namespace,
-    };
-
-    /** @type {!angular.Resource<!backendApi.AppDeploymentSpec>} */
-    let resource = this.resource_('/api/appdeployments');
-
-    this.isDeployInProgress_ = true;
-    resource.save(
-        appDeploymentSpec,
-        (savedConfig) => {
-          this.isDeployInProgress_ = false;
-          this.log_.info('Successfully deployed application: ', savedConfig);
-          this.state_.go(replicasetliststate);
-        },
-        (err) => {
-          this.isDeployInProgress_ = false;
-          this.log_.error('Error deploying application:', err);
-        });
+  deployBySelection() {
+    let deployCtrl = this;
+    deployCtrl.isDeployInProgress = true;
+    deployCtrl.detail.deploy().finally(() => { deployCtrl.isDeployInProgress = false; });
```

Comment:
Use `this.isDeployInProgress` instead of `var foo = this`. Fat arrows have lexical scoping, so this just works. This projects generally uses fat arrows whenever possible and not `var self=this`. The reason is consistency across entire codebase for easier reading.